packages/plugin-stage/sources/drivers/GitDriver.ts

```diff
@@ -1,19 +1,118 @@
 import {execUtils}     from '@berry/core';
-import {NodeFS}        from '@berry/fslib';
+import {NodeFS, xfs}   from '@berry/fslib';
 import {posix}         from 'path';
 
 import * as stageUtils from '../stageUtils';
 
 const MESSAGE_MARKER = `Commit generated via \`yarn stage\``;
 const COMMIT_DEPTH = 11;
 
-async function genCommitMessage(cwd: string) {
+
+async function genCommitMessage(cwd: string, changes: Array<string>, cachedFileLists: string) {
+
+  const updates = new Map();
+  const removes = new Map();
+  const adds = new Map();
+  const createdPackages = [];
+  const removedPackages = [];
+
+
+  for(const change of changes) {
+    const localPath = NodeFS.fromPortablePath(change);
+    const relativePath = posix.relative(cwd, localPath);
+    const doesExistCurrent = xfs.existsSync(localPath);
+    const doesExistPrev = cachedFileLists.indexOf(relativePath) !== -1;
+    if (doesExistCurrent && doesExistPrev) {
+      const {stdout:prevStdout} = await execUtils.execvp(`git`, [`show`, `HEAD~1:${relativePath}`], {cwd, strict: true});
+      const {stdout:currStdout} = await execUtils.execvp(`cat`, [`${localPath}`], {cwd, strict: true});
+      try {
+        const {dependencies: prevDeps, devDependencies: prevDevDeps} = JSON.parse(prevStdout.toString());
+        const {dependencies: currDeps, devDependencies: currDevDeps} = JSON.parse(currStdout.toString());
+        const totalPrevDeps = {...prevDeps, ...prevDevDeps};
+        const totalCurrDeps = {...currDeps, ...currDevDeps};
+        //Find all the  updates/adds/removes on the package.json compare to prev one
+        for(const prevPkg of Object.entries(totalPrevDeps)) {
+          const [name, version] = prevPkg;
+          if(totalCurrDeps[name]) {
+            if (version !== totalCurrDeps[name]) {
+              const key = `Updates ${name} to ${totalCurrDeps[name]}`;
+              setKeyValue(updates, key);
+            }
+          }
+          else {
+            const key = `Removes ${name}`;
+            setKeyValue(removes, key);
+          }
+        }
+
+        for(const currPkg of Object.entries(totalCurrDeps)) {
+          const [name] = currPkg;
+          if (!totalPrevDeps[name]) {
+            const key = `Adds ${name}`;
+            setKeyValue(adds, key);
+          }
+        }
+      }
+      catch(e) {
+        console.log("Error GitDriver", prevStdout, currStdout);
+      }
+    }
+    else if (!doesExistPrev) {
+      //created package.json
+      const {stdout:currStdout} = await execUtils.execvp(`cat`, [`${localPath}`], {cwd, strict: true});
+      const {name: packageName} = JSON.parse(currStdout.toString());
```

Comment:
You want to use `await Manifest.fromFile(localPath)` instead of those two lines. Then by accessing `manifest.name` you get an `Ident`, which you can serialize using `structUtils.stringifyIdent(manifest.name)`.

(Note that it's a bit less performant than just parsing the content and using the `name` field as-is, but I prefer to keep the logic as consistent as possible across the codebase, as long as the perf cost is small)